File: src/main/java/org/jukeboxmc/item/behavior/ItemWaterBucket.java
package org.jukeboxmc.item.behavior;

import org.jukeboxmc.block.Block;
import org.jukeboxmc.block.BlockType;
import org.jukeboxmc.item.ItemType;
import org.jukeboxmc.util.Identifier;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class ItemWaterBucket extends ItemBucket {

    public ItemWaterBucket( Identifier identifier ) {
        super( identifier );
    }

    public ItemWaterBucket( ItemType itemType ) {
        super( itemType );
    }

    @Override
    public Block toBlock() {
        return Block.create( BlockType.WATER );
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/item/behavior/ItemWaterBucket.kt
package org.jukeboxmc.server.item.behavior

import org.jukeboxmc.api.block.Block
import org.jukeboxmc.api.block.BlockType
import org.jukeboxmc.api.item.ItemType

class ItemWaterBucket(itemType: ItemType, countNetworkId: Boolean) : ItemBucket(itemType, countNetworkId) {

    override fun toBlock(): Block {
        return Block.create(BlockType.WATER)
    }

}